### PR TITLE
feat : added explore page functionality

### PR DESCRIPTION
**Pull request by Priya-Patel31**
Fetched all posts .

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import React, { useEffect} from "react";
+import React, { useEffect } from "react";
 import { Routes, Route } from "react-router-dom";
 import Home from "./pages/home/Home";
 import Profile from "./pages/profile/Profile";
```
Comment (t007rushi) [inline src/App.tsx:4]: barrel export here

File: src/App.tsx [hunk 2]
```diff
@@ -27,9 +26,17 @@ function App() {
...
             <PrivateRoute path="/">
               <Home />
-             </PrivateRoute>
+            </PrivateRoute>
           }
         >
+          <Route
+            path="/explore"
+            element={
+              <PrivateRoute>
```
Comment (t007rushi) [inline src/App.tsx:35]: as per naming convention followed you can make it ```RequireAuth```

File: src/app/store.ts [hunk 3]
```diff
@@ -1,5 +1,7 @@
 import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
 import authReducer from "../features/auth/authSlice";
+import exploreReducer from "../features/explore/exploreSlice";
+
 import postsReducer from "../features/posts/PostsSlice";
 import usersReducer from "../features/users/usersSlice";
```
Comment (t007rushi) [inline src/app/store.ts:6]: same make an index file for feature folder

File: src/features/explore/exploreSlice.ts [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+const exploreSlice = createSlice({
+  name: "explore",
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder.addCase(
+      fetchPosts.fulfilled,
+      (state, action: PayloadAction<Post[]>) => {
+        console.log(action.payload);
```
Comment (t007rushi) [inline src/features/explore/exploreSlice.ts:46]: can remove console if not needed

File: src/features/explore/exploreSlice.ts [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+          return post.id !== action.payload.postId;
+        });
+      }
+    );
+    builder.addCase(
+      editPost.fulfilled,
+      (state, action: PayloadAction<ActionPostReturnType>) => {
+        if (
+          !action.payload.explore &&
+          !isPresent({ arr: state.posts, value: action.payload.post.id ?? "" })
```
Comment (t007rushi) [inline src/features/explore/exploreSlice.ts:109]: clean code good work ! keep it @Priya-Patel31

File: src/features/users/usersSlice.ts [hunk 6]
```diff
@@ -22,9 +22,13 @@ export const fetchUsers = createAsyncThunk<User[],string[]>(
     );
    
     const snapshots = await getDocs(q);
+    
     snapshots.forEach((user) => {
       fetchedUsers.push(user.data() as User);
+      console.log(user.data())
```
Comment (t007rushi) [inline src/features/users/usersSlice.ts:28]: same here

**Review by t007rushi — APPROVED** [review]
Very well written 
just  one thing mention above comments future ref something
else good to merge

File: src/features/explore/exploreSlice.ts [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+
+export const fetchPosts = createAsyncThunk<Post[]>(
+  "explore/fetchPosts",
+  async () => {
+    const fetchedPosts: Post[] = [];
+    const q = query(collection(db, "posts"));
+    const snapshots = await getDocs(q);
+    snapshots.forEach((post) => {
+      fetchedPosts.push({ ...post.data(), id: post.id } as Post);
+      console.log(fetchedPosts);
```
Comment (Ankur9669) [inline src/features/explore/exploreSlice.ts:32]: remove console.log

File: src/features/posts/PostsSlice.ts [hunk 8]
```diff
@@ -26,18 +31,21 @@ export const uploadPost = createAsyncThunk<any, Post>(
...
+    return { ...post, id: docId.id };
   }
 );
 
-export const fetchUserPosts = createAsyncThunk<Post[]>(
+export const fetchUserPosts = createAsyncThunk<Post[], string[]>(
   "posts/fetchUserPost",
-  async () => {
+  async (following) => {
+    console.log(following);
```
Comment (Ankur9669) [inline src/features/posts/PostsSlice.ts:41]: fix: remove console.log statement

File: src/pages/explore/Explore.tsx [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+import { useAppDispatch, useAppSelector } from "../../app/hooks";
+import { fetchPosts } from "../../features/explore/exploreSlice";
+import Post from "../../shared/post/Post";
+
+export const Explore = () => {
+  const dispatch = useAppDispatch();
+  const { posts, fetchPostsStatus } = useAppSelector((state) => {
+    return state.explore;
+  });
+  console.log(fetchPostsStatus)
```
Comment (Ankur9669) [inline src/pages/explore/Explore.tsx:11]: fix: remove console

File: src/pages/feed/Feeds.tsx [hunk 10]
```diff
@@ -9,16 +9,21 @@ const Feeds = () => {
   const { posts, fetchPostsStatus } = useAppSelector((state) => {
     return state.posts;
   });
+  const { user } = useAppSelector((state) => {
+    return state.auth;
+  });
   useEffect(() => {
-    if (fetchPostsStatus === "idle") {
-      dispatch(fetchUserPosts());
+    console.log(user?.following);
```
Comment (Ankur9669) [inline src/pages/feed/Feeds.tsx:16]: Same here

File: src/shared/comments/Comments.tsx [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+  MdInsertComment,
+  BsBookmarkFill,
+  BsHeart,
+  BsBookmark,
+} from "../../assets/icons/icons";
+import "./post.css";
+import { Post as PostType } from "../../features/posts/posts.types";
+import { useAppDispatch, useAppSelector } from "../../app/hooks";
+import { bookmarkPost, likePost } from "../../features/posts/PostsSlice";
+
```
Comment (Ankur9669) [inline src/shared/comments/Comments.tsx:16]: fix: Use barrel exports

File: src/shared/comments/Comments.tsx [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+                //   onClick={() => handleMoreOptions()}
+                />
+                {/* {show && (
+                  <MoreOptions post={post} explore={explore}></MoreOptions>
+                )} */}
+              </div>
+            {/* )} */}
+          </div>
+          {/* <div className="caption-container">{post.caption}</div> */}
+        </div>
```
Comment (Ankur9669) [inline src/shared/comments/Comments.tsx:50]: if not required you can remove commented code

File: src/shared/comments/Comments.tsx [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+          {/* {isBookmarked ? <BsBookmarkFill /> : <BsBookmark />} */}
+        </button>
+      </div>
+      <div className="">
+        <div >
+          <img src={Avatar} className="avatar small-avatar" alt="profile" />
+          <div className="flex-col">
+            <input type="text" placeholder="Enter your reply" />
+            <div>
+              <div></div>
```
Comment (Ankur9669) [inline src/shared/comments/Comments.tsx:83]: I dont know if this empty div is there by mistake?

**Review by Ankur9669 — APPROVED** [review]
The functionality and the code both are looking good, I have suggested some changes after that you can merge.

File: src/features/explore/exploreSlice.ts (hunk 5 above)
Comment (Priya-Patel31) [reply src/features/explore/exploreSlice.ts:109]: Thank you !! @t007rushi

File: src/features/explore/exploreSlice.ts (hunk 7 above)
Comment (Priya-Patel31) [reply src/features/explore/exploreSlice.ts:32]: removed

File: src/pages/explore/Explore.tsx (hunk 9 above)
Comment (Priya-Patel31) [reply src/pages/explore/Explore.tsx:11]: removed

**Pull request merged**